### PR TITLE
[Synapse DataPlane]ON BOARD: Synapse DataPlane Sdk Initial Release

### PR DESCRIPTION
**Pull request by aim-for-better**
Synapse is a new on board service. This PR aims for adding data plane sdk for Synapse.
The synapse data plane spec is: [synapse data plane api spec](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview)

Data Plane SDK supports:
- Submit/List/Get/Cancel(Delete) spark batch job
- Submit/List/Get/Cancel(Delete) spark session job
- Submit/List/Get/Cancel(Delete) spark session statement job

### CODE REVIEW

File: sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      CA1032; <!-- Add default exception ctors -->
+      CA1063; <!-- Dispose implementation -->
+      CA1066; <!-- Implement IEquatable -->
+      CA1815; <!-- Equality members on structs -->
+      CA2007; <!-- Do not directly await a Task without calling ConfigureAwait (replaced by AZC0012) -->
+      CA2231; <!-- Override Equality operators -->
+      CA2237; <!-- [Serializeable] on exceptions -->
+      CA2225; <!-- Provide alternative to implicit operators -->
+      CA1714; <!-- Flags should have plural names -->
+    </NoWarn>
```
Comment (AlexGhiondea) [inline sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props]: Any reason to disable all of these?

File: sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props [hunk 2]
```diff
@@ -0,0 +1,133 @@
+﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (AlexGhiondea) [inline sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props]: Why do you need this file?

File: sdk/synapse/Microsoft.Azure.Synapse/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+## Prerequisites
+
+- Microsoft Azure Subscription: To call Microsoft Azure services, including Azure Synapse, you need to first [create an account](https://account.windowsazure.com/Home/Index). If you do not have an existing Azure account, you may sign up for a free trial or use your MSDN subscriber benefits.
+
+- The Azure Synapse client library shares the same [Prerequisites](https://github.com/azure/azure-sdk-for-net#prerequisites) as the Microsoft Azure SDK for .NET.
+
+## Examples
+
+To Do
```
Comment (AlexGhiondea) [inline sdk/synapse/Microsoft.Azure.Synapse/README.md]: Missing examples.

**Review by AlexGhiondea — CHANGES_REQUESTED** [review]
This is all generated code. I have a couple of questions about the non-generated files.

File: sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props (hunk 1 above)
Comment (aim-for-better) [reply sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props]: > Any reason to disable all of these?

Hi @AlexGhiondea , This file is not created by me. I copied the file from eng\Directory.Build.Data.props by following the guide https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#project-structure

File: sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props (hunk 2 above)
Comment (aim-for-better) [reply sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props]: > Why do you need this file?

Please refer the first part for client libraries in the guide https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#project-structure

For short Client library projects needs to use the $(RequiredTargetFrameworks) and refer the client library eventhub I added this file

File: sdk/synapse/Microsoft.Azure.Synapse/README.md (hunk 3 above)
Comment (aim-for-better) [reply sdk/synapse/Microsoft.Azure.Synapse/README.md]: > Missing examples.

Can we add examples after the sdk is published?

File: sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props (hunk 2 above)
Comment (aim-for-better) [reply sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props]: > Why do you need this file?

Removed

File: sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props (hunk 1 above)
Comment (aim-for-better) [reply sdk/synapse/Microsoft.Azure.Synapse/Directory.Build.Data.props]: > Any reason to disable all of these?

Removed the file

**Review by AlexGhiondea — APPROVED** [review]

**Review by chidozieononiwu — APPROVED** [review]
Configuration looks good

**Comment by mitchdenny** [comment]
I am merging this change. The checks that are failing are due to a known build issue with Core. The core pipeline is only running against this change due to a change in ```eng/``` which is unrelated. Merging this in now.

**Pull request merged**